frontend/js/src/user/Dashboard.tsx

```diff
@@ -46,175 +48,79 @@ export type ListensProps = {
-    }
-    if (url.searchParams.get("min_ts")) {
-      minTs = Number(url.searchParams.get("min_ts"));
-    }
-
-    this.setState({ loading: true });
-    const { user } = this.props;
-    const newListens = await this.APIService.getListensForUser(
-      user.name,
-      minTs,
-      maxTs
-    );
-    if (!newListens.length) {
-      // No more listens to fetch
-      if (minTs !== undefined) {
-        this.setState({
-          previousListenTs: undefined,
-        });
-      } else {
-        this.setState({
-          nextListenTs: undefined,
-        });
-      }
-      return;
-    }
-    this.setState(
-      {
-        listens: newListens,
-        lastFetchedDirection: !_.isUndefined(minTs) ? "newer" : "older",
-      },
-      this.afterListensFetch
-    );
-  };
-
-  connectWebsockets = (): void => {
-    this.createWebsocketsConnection();
-    this.addWebsocketsHandlers();
+export default function Listen() {
+  const location = useLocation();
+  const params = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const searchParamsObject = getObjectForURLSearchParams(searchParams);
+
+  const { data } = useQuery({
+    ...RouteQuery(["dashboard", params, searchParamsObject], location.pathname),
+    gcTime: !("max_ts" in searchParamsObject) ? 0 : 1,
+  }) as {
+    data: ListenLoaderData;
   };
 
-  createWebsocketsConnection = (): void => {
+  const {
+    listens: initialListens = [],
+    user,
+    userPinnedRecording = undefined,
+    latestListenTs,
+    oldestListenTs,
+  } = data;
+
+  const previousListenTs = initialListens?.[0]?.listened_at;
+  const nextListenTs = initialListens?.[initialListens.length - 1]?.listened_at;
+
+  const [listens, setListens] = React.useState<Array<Listen>>(
+    initialListens || []
+  );
+
+  const { currentUser, websocketsUrl, APIService } = React.useContext(
+    GlobalAppContext
+  );
+
+  const expectedListensPerPage = 25;
+  const maxWebsocketListens = 7;
+
+  const listensTable = React.createRef<HTMLTableElement>();
+  const socketRef = React.useRef<Socket | null>(null);
+  const [webSocketListens, setWebSocketListens] = React.useState<Array<Listen>>(
+    []
+  );
+  const [followingList, setFollowingList] = React.useState<Array<string>>([]);
+  const [playingNowListen, setPlayingNowListen] = React.useState<
+    Listen | undefined
+  >(
+    initialListens
+      ? _.remove(initialListens, (listen) => listen.playing_now)[0]
+      : undefined
+  );
+  const [deletedListen, setDeletedListen] = React.useState<Listen | null>(null);
+  const [listenCount, setListenCount] = React.useState<number | undefined>();
+  const [dateTimePickerValue, setDateTimePickerValue] = React.useState<Date>(
+    nextListenTs ? new Date(nextListenTs * 1000) : new Date(Date.now())
+  );
+
+  React.useEffect(() => {
+    setListens(initialListens || []);
+  }, [initialListens]);
+
+  const createWebsocketsConnection = React.useCallback((): void => {
```

Comment:
Correct, that the plan for the next PR, since socket is being used by multiple components

Reply:
I mentioned this in our video chat, I'm not so sure anymore that we ant to do that :sweat_smile: 
Basically, we use websocket connections for two pages only at the moment: the user listens page and the playlist page,
In both cases, once the user navigates away to another tab or page, there's really no reason to keep using resources by keeping the websocket connection open.

I can see a future in which we get website notifications show in the UI, or a Playing Now conmponent visible across the website, or something like that, but it might be best to wait until we have that feature in mind.

We also talked with lucifer a little while back about possibly replacing websockets with server-sent events, so maybe we should also revisit that idea before refactoring WS on the frontend.